pkg/operator/operator_test.go

```diff
@@ -176,21 +176,34 @@ func deploymentHasContainer(d *appsv1.Deployment, containerName string) bool {
 	return false
 }
 
-func TestOperatorSyncClusterAPIControllerHealthCheckControllerNotDeployed(t *testing.T) {
```

Comment:
I put the tests under two separete functions so it's obvious which case is tested. Though, if you feel it's better this way, I am fine with that.

Reply:
table driven tests is the recommended approach i guess, https://github.com/golang/go/wiki/TableDrivenTests

Reply:
I find that it depends. Quite often table driven tests end up adding complexity and sometimes I find it easier to do:

```Go
func TestFooDoesX

func TestFooErrorsWithBar

func TestFooSucceedsWithKnownInputs
```